[refresh2020q1][msgpack] - Make error while building msgpack

This error is occuring during the refresh process. I have posted the context of the error and linked the file at the bottom of the issue.

Error:

```
In file included from /hab/cache/src/msgpack-2.1.5/include/msgpack/object.hpp:15,
                 from /hab/cache/src/msgpack-2.1.5/include/msgpack.hpp:10,
                 from /hab/cache/src/msgpack-2.1.5/example/cpp03/custom.cpp:10:
/hab/cache/src/msgpack-2.1.5/include/msgpack/v1/object.hpp: In function 'void msgpack::v1::operator<<(msgpack::v2::object&, const msgpack_object&)':
/hab/cache/src/msgpack-2.1.5/include/msgpack/v1/object.hpp:652:34: error: 'void* memcpy(void*, const void*, size_t)' copying an object of non-trivial type 'struct msgpack::v2::object' from an array of 'const msgpack_object' {aka 'const struct msgpack_object'} [-Werror=class-memaccess]
  652 |     std::memcpy(&o, &v, sizeof(v));
      |                                  ^
In file included from /hab/cache/src/msgpack-2.1.5/include/msgpack/object_fwd.hpp:17,
                 from /hab/cache/src/msgpack-2.1.5/include/msgpack/v1/adaptor/adaptor_base_decl.hpp:14,
                 from /hab/cache/src/msgpack-2.1.5/include/msgpack/adaptor/adaptor_base_decl.hpp:13,
                 from /hab/cache/src/msgpack-2.1.5/include/msgpack/adaptor/adaptor_base.hpp:13,
                 from /hab/cache/src/msgpack-2.1.5/include/msgpack/v1/object_decl.hpp:16,
                 from /hab/cache/src/msgpack-2.1.5/include/msgpack/object_decl.hpp:14,
                 from /hab/cache/src/msgpack-2.1.5/include/msgpack/object.hpp:13,
                 from /hab/cache/src/msgpack-2.1.5/include/msgpack.hpp:10,
                 from /hab/cache/src/msgpack-2.1.5/example/cpp03/custom.cpp:10:
/hab/cache/src/msgpack-2.1.5/include/msgpack/v2/object_fwd.hpp:23:8: note: 'struct msgpack::v2::object' declared here
   23 | struct object : v1::object {
      |        ^~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [example/cpp03/CMakeFiles/custom.dir/build.make:63: example/cpp03/CMakeFiles/custom.dir/custom.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:500: example/cpp03/CMakeFiles/custom.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
   msgpack: Build time: 0m6s
   msgpack: Exiting on error
```

Steps to reproduce:

```
wget https://gist.githubusercontent.com/MindNumbing/255ca89882213bf0a4422f28fb902570/raw/b00ed3638b73d8a697ca411ea5154bd77118d456/hab_studio.env
source hab.env
hab studio enter
build core-plans/msgpack
```

Full Error Log:
[msgpack.txt](https://github.com/habitat-sh/core-plans/files/4455601/msgpack.txt)

--- a/.refresh/remaining_plans_failed.txt
+++ b/.refresh/remaining_plans_failed.txt
@@ -12,7 +12,6 @@ core-plans/ansible
 core-plans/shield-agent
 core-plans/rethinkdb
 core-plans/postfix
-core-plans/msgpack
 core-plans/libksba
 core-plans/libgcrypt
 core-plans/intltool

--- a/.refresh/remaining_plans_passed.txt
+++ b/.refresh/remaining_plans_passed.txt
@@ -296,6 +296,7 @@ core-plans/openvpn
 core-plans/openresty
 core-plans/nmap
 core-plans/netdata
+core-plans/msgpack
 core-plans/mongodb
 core-plans/metricbeat
 core-plans/lzop

--- a/msgpack/plan.sh
+++ b/msgpack/plan.sh
@@ -1,12 +1,12 @@
 pkg_name=msgpack
 pkg_origin=core
-pkg_version="2.1.5"
+pkg_version="3.2.1"
 pkg_maintainer="The Habitat Maintainers <humans@habitat.sh>"
 pkg_description="MessagePack implementation for C and C++"
 pkg_upstream_url="https://github.com/msgpack/msgpack-c"
 pkg_license=('BSL-1.0')
 pkg_source="https://github.com/msgpack/${pkg_name}-c/releases/download/cpp-$pkg_version/${pkg_name}-$pkg_version.tar.gz"
-pkg_shasum="6126375af9b204611b9d9f154929f4f747e4599e6ae8443b337915dcf2899d2b"
+pkg_shasum="433cbcd741e1813db9ae4b2e192b83ac7b1d2dd7968a3e11470eacc6f4ab58d2"
 pkg_deps=(core/glibc)
 pkg_build_deps=(
   core/cmake

